File: src/main/java/com/example/installcertatruntime/cert/TryShortCuts.java
package com.example.installcertatruntime.cert;

import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping(path = "try")
public class TryShortCuts {

    private final TryService tryService;

    public TryShortCuts(TryService tryService) {
        this.tryService = tryService;
    }


    @GetMapping("")
    public String test1() {
        String s1 = "D:\\develop\\installcertatruntime\\src\\main\\java\\com\\example\\installcertatruntime\\cert\\TryShortCuts.java";
        String s = "com/example/install_cert_at_runtime/cert/TryShortCuts.java:21";

        String s2 = concatString(s1, s);

        String s3 = concatString("1", s2);
        String s4 = s3.concat("aaaa");

        tryService.greeting();
        tryService.sayHi();



        return s3;
    }










    private String concatString(String s1, String s) {
        return s1 + s;
    }

}

File: src/main/kotlin/com/example/installcertatruntime/cert/TryShortCuts.kt
package com.example.installcertatruntime.cert

import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("try")
class TryShortCuts(private val tryService: TryService) {

    fun test1(concate: (String, String) -> String): String {
        val s1 = "D:\\develop\\installcertatruntime\\src\\main\\java\\com\\example\\installcertatruntime\\cert\\TryShortCuts.java"
        val s = "com/example/install_cert_at_runtime/cert/TryShortCuts.java:21"
        val s2 = concate(s1, s)

        val s3 = concate("1", s2)
        val s4 = s3 + "aaaa"

        tryService.greeting()
        tryService.sayHi()



        return s3
    }


    fun test2() {
        test1 { s1, s2 -> s1 + s2 }
    }

    private fun concatString(s1: String, s: String): String {
        return s1 + s
    }


}
